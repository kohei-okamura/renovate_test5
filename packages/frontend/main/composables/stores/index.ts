/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { createSessionStore } from '~/composables/stores/create-session-store'
import { NuxtContext } from '~/models/nuxt'

type CreateGlobalStoreParams = Pick<NuxtContext, '$api'>

export const createGlobalStore = (params: CreateGlobalStoreParams) => ({
  session: createSessionStore(params)
})

export type GlobalStore = ReturnType<typeof createGlobalStore>
