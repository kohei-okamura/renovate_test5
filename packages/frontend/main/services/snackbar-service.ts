/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */

import { reactive, toRefs } from '@nuxtjs/composition-api'
import { Refs } from '~/support/reactive'

type Colors = {
  error: undefined
  info: undefined
  success: undefined
  warning: undefined
}

type Color = keyof Colors

export type SnackbarConfig = {
  color?: Color
  text: string
}

type Data = {
  config: SnackbarConfig
}

type SnackbarSugars = {
  readonly [K in keyof Colors]: (text: string) => void
}

export type SnackbarService = Refs<Data> & SnackbarSugars & {
  readonly show: (config: SnackbarConfig) => void
}

export function createSnackbarService (): SnackbarService {
  const colors: Color[] = ['error', 'info', 'success', 'warning']
  const defaultConfig: SnackbarConfig = {
    color: 'info',
    text: ''
  }
  const data = reactive<Data>({
    config: { ...defaultConfig }
  })
  const show = (config: SnackbarConfig): void => {
    data.config = { ...defaultConfig, ...config }
  }
  const sugars = Object.fromEntries(colors.map(color => [color, (text: string) => show({ color, text })]))
  return {
    ...toRefs(data),
    ...sugars as SnackbarSugars,
    show
  }
}
