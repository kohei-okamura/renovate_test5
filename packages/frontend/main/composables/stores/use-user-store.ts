/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { InjectionKey, reactive } from '@nuxtjs/composition-api'
import { assign } from '@zinger/helpers'
import { createStore } from '~/composables/stores/utils'
import { usePlugins } from '~/composables/use-plugins'
import { BankAccount } from '~/models/bank-account'
import { Contract } from '~/models/contract'
import { DwsCertification } from '~/models/dws-certification'
import { DwsProject } from '~/models/dws-project'
import { LtcsInsCard } from '~/models/ltcs-ins-card'
import { LtcsProject } from '~/models/ltcs-project'
import { User } from '~/models/user'
import { UserDwsCalcSpec } from '~/models/user-dws-calc-spec'
import { UserDwsSubsidy } from '~/models/user-dws-subsidy'
import { UserLtcsCalcSpec } from '~/models/user-ltcs-calc-spec'
import { UserLtcsSubsidy } from '~/models/user-ltcs-subsidy'
import { UsersApi } from '~/services/api/users-api'

export const createUserState = () => ({
  bankAccount: undefined as BankAccount | undefined,
  contracts: [] as Contract[],
  dwsCalcSpecs: [] as UserDwsCalcSpec[],
  dwsCertifications: [] as DwsCertification[],
  dwsProjects: [] as DwsProject[],
  dwsSubsidies: [] as UserDwsSubsidy[],
  ltcsCalcSpecs: [] as UserLtcsCalcSpec[],
  ltcsInsCards: [] as LtcsInsCard[],
  ltcsProjects: [] as LtcsProject[],
  ltcsSubsidies: [] as UserLtcsSubsidy[],
  user: undefined as User | undefined
})

export function useUserStore () {
  const { $api } = usePlugins()
  const state = reactive(createUserState())
  const actions = {
    async get (params: UsersApi.GetParams) {
      assign(state, await $api.users.get(params))
    },
    async update ({ form, id }: Parameters<typeof $api.users.update>[0]) {
      assign(state, await $api.users.update({ form, id }))
    },
    async updateBankAccount ({ form }: Parameters<typeof $api.bankAccounts.update>[0]) {
      assign(state, await $api.bankAccounts.update({ form }))
    }
  }
  return createStore({ actions, state })
}

export type UserData = ReturnType<typeof createUserState>

export type UserStore = ReturnType<typeof useUserStore>

export type UserState = UserStore['state']

export const userStoreKey: InjectionKey<UserStore> = Symbol('userStore')

export const userStateKey: InjectionKey<UserState> = Symbol('userState')
