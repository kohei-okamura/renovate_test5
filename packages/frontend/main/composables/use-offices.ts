/*
 * Copyright Â© 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { computed, reactive, toRefs, watch } from '@nuxtjs/composition-api'
import { OfficeQualification } from '@zinger/enums/lib/office-qualification'
import { Permission } from '@zinger/enums/lib/permission'
import { Purpose } from '@zinger/enums/lib/purpose'
import { usePlugins } from '~/composables/use-plugins'
import { OfficeId } from '~/models/office'
import { UserId } from '~/models/user'
import { VSelectOption } from '~/models/vuetify'
import { RefOrValues, unref } from '~/support/reactive'

type BaseParams = {
  isCommunityGeneralSupportCenter?: boolean
  permission?: Permission
  qualifications?: OfficeQualification[]
  userId?: UserId
}
type ParamsWithPurpose = BaseParams & {
  internal?: never
  purpose?: Exclude<Purpose, typeof Purpose.unknown>
}
type ParamsWithInternal = BaseParams & {
  internal: true
  purpose?: never
}
type Params = RefOrValues<ParamsWithPurpose | ParamsWithInternal>

export type OfficeOption = VSelectOption<OfficeId> & {
  keyword: string
}

const createOfficeResolver = (xs: VSelectOption<OfficeId>[]) => {
  return (id: OfficeId | undefined, alternative = '-') => (id && xs.find(x => x.value === id)?.text) ?? alternative
}

export const useOffices = (params: Params) => {
  const { $api } = usePlugins()
  const state = reactive({
    isLoadingOffices: true,
    officeOptions: [] as OfficeOption[]
  })
  const resolveOfficeAbbr = computed(() => createOfficeResolver(state.officeOptions))
  watch(
    computed(() => ({
      isCommunityGeneralSupportCenter: unref(params.isCommunityGeneralSupportCenter),
      internal: unref(params.internal),
      permission: unref(params.permission),
      purpose: unref(params.purpose),
      qualifications: unref(params.qualifications),
      userId: unref(params.userId)
    })),
    async ({ internal, purpose, ...rest }) => {
      state.isLoadingOffices = true
      state.officeOptions = await $api.options.offices({ ...rest, purpose: internal ? Purpose.internal : purpose })
      state.isLoadingOffices = false
    },
    { immediate: true }
  )
  return {
    ...toRefs(state),
    resolveOfficeAbbr
  }
}
