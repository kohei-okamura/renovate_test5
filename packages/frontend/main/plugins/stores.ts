/*
 * Copyright Â© 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { defineNuxtPlugin, InjectionKey, onGlobalSetup, provide } from '@nuxtjs/composition-api'
import {
  attendancesStateKey,
  attendancesStoreKey,
  useAttendancesStore
} from '~/composables/stores/use-attendances-store'
import {
  billingBulkUpdateStateKey,
  billingBulkUpdateStoreKey,
  useBillingBulkUpdateStore
} from '~/composables/stores/use-billing-bulk-update-store'
import {
  dwsBillingsStateKey,
  dwsBillingsStoreKey,
  useDwsBillingsStore
} from '~/composables/stores/use-dws-billings-store'
import {
  dwsProvisionReportsIndexStateKey,
  dwsProvisionReportsIndexStoreKey,
  useDwsProvisionReportsStore
} from '~/composables/stores/use-dws-provision-reports-store'
import {
  ltcsBillingsStateKey,
  ltcsBillingsStoreKey,
  useLtcsBillingsStore
} from '~/composables/stores/use-ltcs-billings-store'
import {
  ltcsProvisionReportsIndexStateKey,
  ltcsProvisionReportsIndexStoreKey,
  useLtcsProvisionReportsStore
} from '~/composables/stores/use-ltcs-provision-reports-store'
import {
  notificationStateKey,
  notificationStoreKey,
  useNotificationStore
} from '~/composables/stores/use-notification-store'
import { officesStateKey, officesStoreKey, useOfficesStore } from '~/composables/stores/use-offices-store'
import {
  ownExpenseProgramsStateKey,
  ownExpenseProgramsStoreKey,
  useOwnExpenseProgramsStore
} from '~/composables/stores/use-own-expense-programs-store'
import { rolesStateKey, rolesStoreKey, useRolesStore } from '~/composables/stores/use-roles-store'
import { sessionStateKey, sessionStoreKey, useSessionStore } from '~/composables/stores/use-session-store'
import {
  shiftImportsStateKey,
  shiftImportsStoreKey,
  useShiftImportsStore
} from '~/composables/stores/use-shift-imports-store'
import { shiftsStateKey, shiftsStoreKey, useShiftsStore } from '~/composables/stores/use-shifts-store'
import { staffsStateKey, staffsStoreKey, useStaffsStore } from '~/composables/stores/use-staffs-store'
import {
  userBillingsStateKey,
  userBillingsStoreKey,
  useUserBillingsStore
} from '~/composables/stores/use-user-billings-store'
import { usersStateKey, usersStoreKey, useUsersStore } from '~/composables/stores/use-users-store'
import { StateProvider } from '~/composables/stores/utils'

type ProvideStore = <T extends Record<string, unknown>, U extends StateProvider<T>> (
  store: U,
  storeKey: InjectionKey<U>,
  stateKey: InjectionKey<T>
) => void

const provideStore: ProvideStore = (store, storeKey, stateKey) => {
  provide(storeKey, store)
  provide(stateKey, store.state)
}

export default defineNuxtPlugin(() => onGlobalSetup(() => {
  provideStore(useUserBillingsStore(), userBillingsStoreKey, userBillingsStateKey)
  provideStore(useLtcsBillingsStore(), ltcsBillingsStoreKey, ltcsBillingsStateKey)
  provideStore(useDwsBillingsStore(), dwsBillingsStoreKey, dwsBillingsStateKey)
  provideStore(useAttendancesStore(), attendancesStoreKey, attendancesStateKey)
  provideStore(useShiftsStore(), shiftsStoreKey, shiftsStateKey)
  provideStore(useOwnExpenseProgramsStore(), ownExpenseProgramsStoreKey, ownExpenseProgramsStateKey)
  provideStore(useOfficesStore(), officesStoreKey, officesStateKey)
  provideStore(useStaffsStore(), staffsStoreKey, staffsStateKey)
  provideStore(useUsersStore(), usersStoreKey, usersStateKey)
  provideStore(useDwsProvisionReportsStore(), dwsProvisionReportsIndexStoreKey, dwsProvisionReportsIndexStateKey)
  provideStore(useLtcsProvisionReportsStore(), ltcsProvisionReportsIndexStoreKey, ltcsProvisionReportsIndexStateKey)
  provideStore(useNotificationStore(), notificationStoreKey, notificationStateKey)
  provideStore(useRolesStore(), rolesStoreKey, rolesStateKey)
  provideStore(useSessionStore(), sessionStoreKey, sessionStateKey)
  provideStore(useShiftImportsStore(), shiftImportsStoreKey, shiftImportsStateKey)
  provideStore(useBillingBulkUpdateStore(), billingBulkUpdateStoreKey, billingBulkUpdateStateKey)
}))
