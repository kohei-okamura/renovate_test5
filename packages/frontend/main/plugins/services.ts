/*
 * Copyright Â© 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { defineNuxtPlugin } from '@nuxtjs/composition-api'
import { createGlobalStore, GlobalStore } from '~/composables/stores'
import { AlertService, createAlertService } from '~/services/alert-service'
import { ApiService, createApiService } from '~/services/api-service'
import { BreadcrumbsService, createBreadcrumbsService } from '~/services/breadcrumbs-service'
import { ConfirmDialogService, createConfirmDialogService } from '~/services/confirm-dialog-service'
import { createDownloadService, DownloadService } from '~/services/download-service'
import { createDrawerService, DrawerService } from '~/services/drawer-service'
import { createFormService, FormService } from '~/services/form-service'
import { createRoutesService, RoutesService } from '~/services/routes-service'
import { createSnackbarService, SnackbarService } from '~/services/snackbar-service'
import { createTabService, TabService } from '~/services/tab-service'

export type Services = Readonly<{
  $alert: AlertService
  $api: ApiService
  $breadcrumbs: BreadcrumbsService
  $confirm: ConfirmDialogService
  $download: DownloadService
  $drawer: DrawerService
  $form: FormService
  $globalStore: GlobalStore
  $routes: RoutesService
  $snackbar: SnackbarService
  $tabs: TabService
}>

export default defineNuxtPlugin((context, inject) => {
  inject('alert', createAlertService(context))
  inject('api', createApiService(context))
  inject('breadcrumbs', createBreadcrumbsService())
  inject('confirm', createConfirmDialogService())
  inject('download', createDownloadService())
  inject('drawer', createDrawerService())
  inject('form', createFormService(context))
  inject('globalStore', createGlobalStore(context))
  inject('routes', createRoutesService(context))
  inject('snackbar', createSnackbarService())
  inject('tabs', createTabService())
})
