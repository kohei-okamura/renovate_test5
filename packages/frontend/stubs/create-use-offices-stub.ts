/*
 * Copyright Â© 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { computed, reactive, toRefs } from '@nuxtjs/composition-api'
import { useOffices } from '~/composables/use-offices'
import { Office, OfficeId } from '~/models/office'
import { createOfficeStubs } from '~~/stubs/create-office-stub'

export const createUseOfficesStub = (offices?: Office[]): ReturnType<typeof useOffices> => {
  const options = (offices ?? createOfficeStubs()).map(x => ({ value: x.id, text: x.abbr, keyword: x.name }))
  const resolveOfficeAbbr = computed(() => (id: OfficeId | undefined, alternative = '-') => {
    return (id && options.find(x => x.value === id)?.text) ?? alternative
  })
  const data = reactive({
    isLoadingOffices: false,
    officeOptions: options
  })
  return {
    ...toRefs(data),
    resolveOfficeAbbr
  }
}
