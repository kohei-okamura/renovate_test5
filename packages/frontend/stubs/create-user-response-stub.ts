/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { UserId } from '~/models/user'
import { UsersApi } from '~/services/api/users-api'
import { createBankAccountStub } from '~~/stubs/create-bank-account-stub'
import { createContractStubsForUser } from '~~/stubs/create-contract-stub'
import { createDwsCertificationStubsForUser } from '~~/stubs/create-dws-certification-stub'
import { createDwsProjectStubs } from '~~/stubs/create-dws-project-stub'
import { createDwsSubsidiesStub } from '~~/stubs/create-dws-subsidy-stub'
import { createLtcsInsCardStubsForUser } from '~~/stubs/create-ltcs-ins-card-stub'
import { createLtcsProjectStubs } from '~~/stubs/create-ltcs-project-stub'
import { createLtcsSubsidyStubsForUser } from '~~/stubs/create-ltcs-subsidy-stub'
import { createUserDwsCalcSpecStubsForUser } from '~~/stubs/create-user-dws-calc-spec-stub'
import { createUserLtcsCalcSpecStubsForUser } from '~~/stubs/create-user-ltcs-calc-spec-stub'
import { createUserStub } from '~~/stubs/create-user-stub'

export function createUserResponseStub (id: UserId = 1): UsersApi.GetResponse {
  const user = createUserStub(id)
  const contracts = createContractStubsForUser(id)
  return {
    bankAccount: createBankAccountStub(user.bankAccountId),
    contracts,
    dwsCertifications: createDwsCertificationStubsForUser(user.id),
    dwsProjects: createDwsProjectStubs(contracts),
    dwsSubsidies: createDwsSubsidiesStub(user.id),
    dwsCalcSpecs: createUserDwsCalcSpecStubsForUser(user.id),
    ltcsInsCards: createLtcsInsCardStubsForUser(user.id),
    ltcsProjects: createLtcsProjectStubs(contracts),
    ltcsSubsidies: createLtcsSubsidyStubsForUser(user.id),
    ltcsCalcSpecs: createUserLtcsCalcSpecStubsForUser(user.id),
    user
  }
}
