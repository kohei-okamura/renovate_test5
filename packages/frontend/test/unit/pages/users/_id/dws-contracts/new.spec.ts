/*
 * Copyright © 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { Wrapper } from '@vue/test-utils'
import { ServiceSegment } from '@zinger/enums/lib/service-segment'
import { camelToKebab, noop } from '@zinger/helpers'
import { createMock } from '@zinger/helpers/testing/create-mock'
import { mocked } from '@zinger/helpers/testing/mocked'
import Vue from 'vue'
import { userStateKey, userStoreKey } from '~/composables/stores/use-user-store'
import { useOffices } from '~/composables/use-offices'
import { HttpStatusCode } from '~/models/http-status-code'
import DwsContractNewPage from '~/pages/users/_id/dws-contracts/new.vue'
import { DwsContractsApi } from '~/services/api/dws-contracts-api'
import { SnackbarService } from '~/services/snackbar-service'
import { createUseOfficesStub } from '~~/stubs/create-use-offices-stub'
import { createUserResponseStub } from '~~/stubs/create-user-response-stub'
import { createUserStoreStub } from '~~/stubs/create-user-store-stub'
import { createAxiosError } from '~~/test/helpers/create-axios-error'
import { createMockedApi } from '~~/test/helpers/create-mocked-api'
import { createMockedFormService } from '~~/test/helpers/create-mocked-form-service'
import { createMockedRoute } from '~~/test/helpers/create-mocked-route'
import { createMockedRouter } from '~~/test/helpers/create-mocked-router'
import { provides } from '~~/test/helpers/provides'
import { setupComponentTest } from '~~/test/helpers/setup-component-test'

jest.mock('~/composables/use-offices')

describe('pages/users/_id/dws-contracts/new.vue', () => {
  const { mount } = setupComponentTest()
  const $api = createMockedApi('dwsContracts', 'users')
  const $router = createMockedRouter()
  const $route = createMockedRoute({
    query: { segment: `${ServiceSegment.disabilitiesWelfare}` }
  })
  const $snackbar = createMock<SnackbarService>()
  const $form = createMockedFormService()
  const form: DwsContractsApi.CreateForm = {
    officeId: 3,
    note: 'だるまさんがころんだ'
  }
  const mocks = {
    $api,
    $form,
    $route,
    $router,
    $snackbar
  }
  const userId = 1

  let wrapper: Wrapper<Vue & any>

  function mountComponent () {
    const response = createUserResponseStub(userId)
    const userStore = createUserStoreStub(response)
    wrapper = mount(DwsContractNewPage, {
      ...provides(
        [userStateKey, userStore.state],
        [userStoreKey, userStore]
      ),
      mocks
    })
  }

  function unmountComponent () {
    wrapper.destroy()
  }

  beforeAll(() => {
    mocked(useOffices).mockReturnValue(createUseOfficesStub())
  })

  afterAll(() => {
    mocked(useOffices).mockReset()
  })

  it('should be rendered correctly', () => {
    mountComponent()
    expect(wrapper).toMatchSnapshot()
    unmountComponent()
  })

  describe('submit', () => {
    // scrollIntoViewが定義されてないエラーを回避するために空の関数にする
    // 参考:https://github.com/jsdom/jsdom/issues/1695
    Element.prototype.scrollIntoView = noop

    beforeAll(() => {
      mountComponent()
    })

    afterAll(() => {
      unmountComponent()
    })

    beforeEach(() => {
      jest.spyOn($api.dwsContracts, 'create').mockResolvedValue(undefined)
      jest.spyOn($snackbar, 'success').mockReturnValue()
      jest.spyOn($snackbar, 'error').mockReturnValue()
    })

    afterEach(() => {
      mocked($snackbar.success).mockReset()
      mocked($snackbar.error).mockReset()
      mocked($api.dwsContracts.create).mockReset()
    })

    it('should call $api.dwsContracts.create when pass the validation', async () => {
      await wrapper.vm.submit(form)

      expect($api.dwsContracts.create).toHaveBeenCalledTimes(1)
      expect($api.dwsContracts.create).toHaveBeenCalledWith({ form, userId })
    })

    it('should display message when succeeded', async () => {
      await wrapper.vm.submit(form)

      expect($snackbar.success).toHaveBeenCalledTimes(1)
      expect($snackbar.success).toHaveBeenCalledWith('契約情報を登録しました。')
    })

    it.each([
      ['officeId', '事業所を入力してください。']
    ])(
      'should display errors when server responses 400 Bad Request (error occurred in "%s")',
      async (key, message) => {
        jest.spyOn($api.dwsContracts, 'create').mockRejectedValue(createAxiosError(HttpStatusCode.BadRequest, {
          errors: {
            [key]: [message]
          }
        }))

        await wrapper.vm.submit(form)
        await wrapper.vm.$nextTick()
        await jest.runAllTimers()

        const targetWrapper = wrapper.find(`[data-${camelToKebab(key)}]`)

        expect($snackbar.success).not.toHaveBeenCalled()
        expect($snackbar.error).toHaveBeenCalledTimes(1)
        expect($snackbar.error).toHaveBeenCalledWith('正しく入力されていない項目があります。入力内容をご確認ください。')
        expect(targetWrapper.text()).toContain(message)
        expect(targetWrapper).toMatchSnapshot()
      }
    )
  })
})
