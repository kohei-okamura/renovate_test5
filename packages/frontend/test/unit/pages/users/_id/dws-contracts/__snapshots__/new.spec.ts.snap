// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/users/_id/dws-contracts/new.vue should be rendered correctly 1`] = `
<div
  breadcrumbs="[object Object],[object Object],[object Object]"
  class="z-page-stub"
  compact="compact"
>
  <div
    class="z-page-stub-slot-tabs"
  />
  <div
    class="z-page-stub-slot-title"
  />
  <form
    class="v-form"
    data-form=""
    novalidate="novalidate"
  >
    <div>
      <z-user-summary-stub
        user="[object Object]"
      />
      <div
        class="z-form-card-stub"
        title="契約情報"
      >
        <div
          class="z-form-card-stub-slot-actions"
        />
        <div
          class="root mt-3"
          data-contracted-on=""
          data-z-form-card-item=""
        />
        <z-data-card-item-stub
          icon="M11.19,2.25C10.93,2.25 10.67,2.31 10.42,2.4L3.06,5.45C2.04,5.87 1.55,7.04 1.97,8.05L6.93,20C7.24,20.77 7.97,21.23 8.74,21.25C9,21.25 9.27,21.22 9.53,21.1L16.9,18.05C17.65,17.74 18.11,17 18.13,16.25C18.14,16 18.09,15.71 18,15.45L13,3.5C12.71,2.73 11.97,2.26 11.19,2.25M14.67,2.25L18.12,10.6V4.25A2,2 0 0,0 16.12,2.25M20.13,3.79V12.82L22.56,6.96C22.97,5.94 22.5,4.78 21.47,4.36M11.19,4.22L16.17,16.24L8.78,19.3L3.8,7.29"
          label="事業領域"
          value="障害福祉サービス"
        />
        <z-data-card-item-stub
          icon="M12,2A10,10 0 0,1 22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2M12,4A8,8 0 0,0 4,12A8,8 0 0,0 12,20A8,8 0 0,0 20,12A8,8 0 0,0 12,4M12,10.5A1.5,1.5 0 0,1 13.5,12A1.5,1.5 0 0,1 12,13.5A1.5,1.5 0 0,1 10.5,12A1.5,1.5 0 0,1 12,10.5M7.5,10.5A1.5,1.5 0 0,1 9,12A1.5,1.5 0 0,1 7.5,13.5A1.5,1.5 0 0,1 6,12A1.5,1.5 0 0,1 7.5,10.5M16.5,10.5A1.5,1.5 0 0,1 18,12A1.5,1.5 0 0,1 16.5,13.5A1.5,1.5 0 0,1 15,12A1.5,1.5 0 0,1 16.5,10.5Z"
          label="契約状態"
        >
          <!---->
          <span>
            仮契約
          </span>
        </z-data-card-item-stub>
        <z-form-card-item-set-stub
          icon="M18,15H16V17H18M18,11H16V13H18M20,19H12V17H14V15H12V13H14V11H12V9H20M10,7H8V5H10M10,11H8V9H10M10,15H8V13H10M10,19H8V17H10M6,7H4V5H6M6,11H4V9H6M6,15H4V13H6M6,19H4V17H6M12,7V3H2V21H22V7H12Z"
          label=""
        >
          <div
            class="root"
            data-office-id=""
            data-z-form-card-item=""
          >
            <div
              class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-select v-autocomplete"
              error-messages=""
              label="事業所 *"
            >
              <div
                class="v-input__control"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  class="v-input__slot"
                  role="combobox"
                  style="background-color: rgb(241, 246, 250); border-color: #f1f6fa;"
                >
                  <fieldset
                    aria-hidden="true"
                  >
                    <legend
                      style="width: 0px;"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                  <div
                    class="v-select__slot"
                  >
                    <label
                      class="v-label theme--light"
                      style="left: 0px; position: absolute;"
                    >
                      事業所 *
                    </label>
                    <input
                      autocomplete="off"
                      type="text"
                    />
                    <div
                      class="v-input__append-inner"
                    >
                      <div
                        class="v-input__icon v-input__icon--append"
                      >
                        <i
                          aria-hidden="true"
                          class="v-icon notranslate mdi mdi-menu-down theme--light"
                        />
                      </div>
                    </div>
                    <input
                      type="hidden"
                    />
                  </div>
                  <div
                    class="v-menu"
                  >
                    <!---->
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <transition-group-stub
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </z-form-card-item-set-stub>
        <!---->
        <z-form-card-item-set-stub
          icon="M21,6V8H3V6H21M3,18H12V16H3V18M3,13H21V11H3V13Z"
          label=""
        >
          <div
            class="root"
            data-note=""
            data-z-form-card-item=""
          >
            <div
              class="v-input v-textarea v-textarea--auto-grow v-textarea--no-resize theme--light v-text-field v-text-field--enclosed v-text-field--outlined"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                  style="background-color: rgb(241, 246, 250); border-color: #f1f6fa;"
                >
                  <fieldset
                    aria-hidden="true"
                  >
                    <legend
                      style="width: 0px;"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label theme--light"
                      style="left: 0px; position: absolute;"
                    >
                      備考
                    </label>
                    <textarea
                      rows="3"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light"
                  >
                    <transition-group-stub
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                  <div
                    class="v-counter theme--light"
                  >
                    0 / 255
                  </div>
                </div>
              </div>
            </div>
          </div>
        </z-form-card-item-set-stub>
        <div
          class="z-form-card-stub-slot-header"
        />
      </div>
      <!---->
      <z-form-action-button-stub
        icon="M14,13V17H10V13H7L12,8L17,13M19.35,10.03C18.67,6.59 15.64,4 12,4C9.11,4 6.6,5.64 5.35,8.03C2.34,8.36 0,10.9 0,14A6,6 0 0,0 6,20H19A5,5 0 0,0 24,15C24,12.36 21.95,10.22 19.35,10.03Z"
        text="登録"
      />
    </div>
  </form>
</div>
`;

exports[`pages/users/_id/dws-contracts/new.vue submit should display errors when server responses 400 Bad Request (error occurred in "officeId") 1`] = `
<div
  class="root"
  data-office-id=""
  data-z-form-card-item=""
>
  <div
    class="v-input v-input--has-state v-input--dense theme--light v-text-field v-text-field--is-booted v-text-field--enclosed v-text-field--outlined v-select v-autocomplete error--text"
    error-messages="事業所を入力してください。"
    label="事業所 *"
  >
    <div
      class="v-input__control"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        class="v-input__slot"
        role="combobox"
        style="background-color: rgb(241, 246, 250); border-color: #f1f6fa;"
      >
        <fieldset
          aria-hidden="true"
        >
          <legend
            style="width: 0px;"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
        <div
          class="v-select__slot"
        >
          <label
            class="v-label theme--light error--text"
            style="left: 0px; position: absolute;"
          >
            事業所 *
          </label>
          <input
            autocomplete="off"
            type="text"
          />
          <div
            class="v-input__append-inner"
          >
            <div
              class="v-input__icon v-input__icon--append"
            >
              <i
                aria-hidden="true"
                class="v-icon notranslate mdi mdi-menu-down theme--light error--text"
              />
            </div>
          </div>
          <input
            type="hidden"
          />
        </div>
        <div
          class="v-menu"
        >
          <!---->
        </div>
      </div>
      <div
        class="v-text-field__details"
      >
        <div
          class="v-messages theme--light error--text"
          role="alert"
        >
          <transition-group-stub
            class="v-messages__wrapper"
            name="message-transition"
            tag="div"
          >
            <div
              class="v-messages__message"
            >
              事業所を入力してください。
            </div>
          </transition-group-stub>
        </div>
      </div>
    </div>
  </div>
</div>
`;
